vpr/src/timing/timing_graph_builder.cpp

```diff
@@ -351,7 +596,7 @@ void TimingGraphBuilder::add_block_to_timing_graph(const AtomBlockId blk) {
                 NodeId sink_tnode = netlist_lookup_.atom_pin_tnode(sink_pin, BlockTnode::EXTERNAL);
 
                 tg_->add_edge(tatum::EdgeType::PRIMITIVE_COMBINATIONAL, src_tnode, sink_tnode);
-                VTR_LOG("Adding edge from '%s' (%zu) -> '%s' (%zu)\n", netlist_.pin_name(src_clock_pin).c_str(), size_t(src_tnode), netlist_.pin_name(sink_pin).c_str(), size_t(sink_tnode));
+                VTR_LOG("Adding edge from '%s' (tnode: %zu) -> '%s' (tnode: %zu) to allow clocks to propagate\n", netlist_.pin_name(src_clock_pin).c_str(), size_t(src_tnode), netlist_.pin_name(sink_pin).c_str(), size_t(sink_tnode));
```

Comment:
It would be good to add messages referencing pin names to all assert messages wherever possible. A simple message that there is something wrong with the timing graph does not tell much without the information where the problem is in the architecture / netlist.

Reply:
> A simple message that there is something wrong with the timing graph does not tell much without the information where the problem is in the architecture / netlist.

I agree.

> It would be good to add messages referencing pin names to all assert messages wherever possible.

However, I'm less concerned about having end-user friendly assert messages (which ideally should only be relevant to developers), and more concerned that we should produce good error messages.

If an assertion is regularly firing due to bad user input, that means it should be converted to an error with an appropriate message (including user-friendly information).